### ISSUE TITLE
Move state-definition outside of component.

### ISSUE THREAD

**Issue by AlphaMoonbaseBerlin** [rosofo/transit#10]
Right now you need to edit the Extension itself to change the state-network, which is against the best practive of seperating data and logic.
The upasited of having the data (in this case the state definiton) seperate is simple: It allows for rapid switching of state-definitons and to update the transitCOMP without lossing the data itself. 
In your case I would handle it using callbacks, either using the TD-Callback system (https://docs.derivative.ca/CallbacksExt_Examples) or my callbackManager. 
You can either pass the mashine directly as an argument or pass the module. i.E. the __ini__ becomes:
```python
self.transit = Transit(self)
self.Machine = self.DoCallback( "onInit", Machine )
self.transit.expose_machine(self.Machine)
``` 
and the callback looks like this:
```python
def onInit(Machine):
    return Machine(
        self,
        name="myco",
        states=["smooth", "chaotic"],
        initial="smooth",
        transitions=[["go", "smooth", "chaotic"], ["go", "chaotic", "smooth"]],
    )
```
This way we create a nice wrapper arround the library which is updateable.

**Comment by rosofo** [comment]
Yes this makes sense, will make this part of the next version 👍

**Comment by rosofo** [comment]
implemented as of 48fe585
